### PR TITLE
[RDPHOEN-1165] fitimage for netboot

### PR DESCRIPTION
**Pull request by iris-ersc**
Fix nfsboot.
## How to test
Build any uuu image
```
bitbake mc:imx8mp-evk:irma6-base-uuu
```

Flash generated flashalll.uuu. Unpack `rootfs ... tar.bz2` to your nfs location.

Build netboot fitimage
```
bitbake mc:imx8mp-evk:irma6-fitimage-netboot
```

Copy `irma6-fitimage-netboot.itb.signed` to your tftp location, rename to `fitImage.signed`.

Boot board, check if mmcboot is working. Reboot to U-Boot console. `setenv nfsroot` to your nfs location. Run nfsboot with
```
run netfitboot
```

If you don't have a dhcp server running, also `setenv serverip` and replace `dhcp` with `tftp` in the `netfitboot` variable.

### CODE REVIEW

File: recipes-core/initrdscripts/files/initramfs-init.sh [hunk 1]
```diff
@@ -108,6 +107,8 @@ DATA_DEV="/dev/mapper/irma6lvm-userdata"
 DECRYPT_DATA_NAME="decrypted-irma6lvm-userdata"
 
 
+debug "Select ${FIRMWARE_SUFFIX}"
```
Comment (linkjumper) [inline recipes-core/initrdscripts/files/initramfs-init.sh:95]: The output ld be: "Select _a" or "Select _b".

File: recipes-core/images/irma6-initramfs.bb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# SPDX-License-Identifier: MIT
+# Copyright (C) 2022 iris-GmbH infrared & intelligent sensors
+
+# use recipe from meta-secure-imx as base
+require recipes-core/images/crypt-image-initramfs.inc
+
+INITRAMFS_COMMON_PACKAGES_remove = " \
+    u-boot-env \
+    libubootenv-bin \
```
Comment (linkjumper) [inline recipes-core/images/irma6-initramfs.bb:9]: Are you sure we do not need lvm2 and libubootenv-bin?

File: recipes-core/images/irma6-initramfs-netboot.bb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+INITRAMFS_COMMON_PACKAGES_remove = " \
+    u-boot-env \
+    libubootenv-bin \
+    initramfs-init \
+"
+
+INITRAMFS_COMMON_PACKAGES_append = " \
+    irma6-initrdscripts-netboot \
+"
+
```
Comment (linkjumper) [inline recipes-core/images/irma6-initramfs-netboot.bb:16]: What do you think about combining the equal thinks in irma6-initramfs.inc from irma6-initramfs.bb and irma6-initramfs-netboot.bb. It is btw. not bad having the caam tools in the nfs iniramfs, is it?

File: recipes-core/images/irma6-fitimage-netboot.bb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+SUMMARY = "IRMA6 FIT image with initramfs"
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"
+
+FITIMAGE_DESCRIPTION = "IRMA6 default fitimage for netboot"
+FITIMAGE_RUNNING_VERSION = "1.0"
+
+# name of the recipe, which creates the initramfs
+RESCUE_NAME = "irma6-initramfs-netboot"
+
```
Comment (linkjumper) [inline recipes-core/images/irma6-fitimage-netboot.bb:13]: The irma6 prefix everywhere is killing me :dagger: :D

**Review by linkjumper — CHANGES_REQUESTED** [review]

**Comment by linkjumper** [comment]
Tested on imx8mp-evk with irma6-maintenance image!

**Comment by Eximmius** [comment]
What do you think of `bitbake mc:imx8mp-evk:irma6-**maintenance**-netboot` instead of `bitbake mc:imx8mp-evk:irma6-**fitimage**-netboot` ?

File: recipes-core/images/irma6-initramfs.bb (hunk 2 above)
Comment (iris-ersc) [reply recipes-core/images/irma6-initramfs.bb]: Those are the ones, that were not included in the previous bbappend recipe. So, no change basically

File: recipes-core/images/irma6-initramfs-netboot.bb (hunk 3 above)
Comment (iris-ersc) [reply recipes-core/images/irma6-initramfs-netboot.bb]: inc-file created. I also moved the rdepends on the caam tools into the initscript recipe

**Review by linkjumper — APPROVED** [review]

**Pull request merged**